common/templates/context_funcs.go

```diff
@@ -277,12 +277,24 @@ func (c *Context) checkSafeStringDictNoRecursion(d SDict, n int) bool {
 			}
 		}
 
+		if cast, ok := v.(*Dict); ok {
+			if !c.checkSafeDictNoRecursion(*cast, n+1) {
+				return false
+			}
+		}
+
```

Comment:
This whole block of code would probably be better written as a type switch given the number of type casts, but I didn't want to accidentally cause more breakage : )